libs/common/src/sql-db-entities/payment/payment.entity.ts

```diff
@@ -51,4 +54,26 @@ export class PaymentEntity extends AuditableBaseEntity {
     payment.updateLastUpdatedBy(requestSource);
     return payment;
   }
+
+  public feeRecordsMatchPayment(): boolean {
+    if (this.feeRecords.length === 0) {
+      throw new PaymentHasNoFeeRecordsError();
+    }
+
+    if (this.feeRecords.some(({ paymentCurrency }) => paymentCurrency !== this.currency)) {
+      throw new PaymentAndFeeRecordCurrencyDoesNotMatchError();
+    }
```

Comment:
I feel a bit conflicted about these errors, it's a bit hard to tell without seeing the intended usage, but I would say we should never have a payment not attached to any fee records, and we should never have a payment that doesn't match the fee record payment currency. Any request to that effect should be rejected before we ever create an entity no? As such, it seems bizarre to be checking for that and maintaining specific error codes for those cases that are currently only getting thrown from here?

Reply:
I've read the ticket and see why these checks have been added. That said, I don't really like that the ticket this comes from has no ACs, I think it would make more sense for these checks to be added down the line, if and when it becomes possible for these situations to occur?